plugins/polio/api/vaccines/repository.py

```diff
@@ -0,0 +1,260 @@
+from datetime import datetime
+
+from django.db.models import Q, Min, Max, OuterRef, Subquery
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import cache_page
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import permissions, serializers, filters
+from rest_framework.filters import OrderingFilter, SearchFilter
+from rest_framework.mixins import ListModelMixin
+from rest_framework.viewsets import GenericViewSet
+from datetime import timedelta
+
+from iaso.api.common import Paginator
+from plugins.polio.models import (
+    Campaign,
+    CampaignType,
+    DestructionReport,
+    IncidentReport,
+    OutgoingStockMovement,
+    Round,
+    VaccinePreAlert,
+    VaccineRequestForm,
+    VaccineRequestFormType,
+)
+
```

Comment:
There is a few unused imports here.

I think I'm gonna add `ruff` to the project because this can be checked and fixed automatically.